lib/Models/CsvCatalogItem.js

```diff
@@ -562,6 +582,56 @@ CsvCatalogItem.prototype.activateColumnFromTableStyle = function() {
     }
 };
 
+
+/**
+ * Loads data from a URL into a (usually temporary) table structure.
+ * @param  {String} url The URL.
+ * @return {Promise} A promise which resolves to a table structure.
+ */
+function loadIntoTableStructure(url) {
+    // Load in the data file as a TableStructure. Currently only understands csv.
+    const tableStructure = new TableStructure();
+    return loadText(url).then(tableStructure.loadFromCsv.bind(tableStructure));
```

Comment:
This needs to go through the proxy if so configured.  Use the '0d' form when polling so it doesn't cache.